dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/persister/EventPersister.java

```diff
@@ -215,4 +215,11 @@ protected void persistOwnership( TrackerPreheat preheat, ProgramStageInstance en
     {
         // DO NOTHING. Event creation does not create ownership records.
     }
+
+    @Override
+    protected String getUpdatedTrackedEntity( ProgramStageInstance entity )
+    {
+        return Optional.ofNullable( entity.getProgramInstance() )
+            .map( pi -> pi.getEntityInstance().getUid() ).orElse( null );
```

Comment:
If it is a programEvent, programInstance is defined but TEI is null, so this can throw a NPE

Reply:
ok good point thanks